update dependencies

Locke opened the pull request:
The first TravisCI job for #125 failed to download a dependency. Thereby I noticed that some dependencies are outdated and checked them all.

I did not update:

* cats / eff / diffson: compile error
* circe: mentions "it's worth taking a look at the newly-clarified semantics [..] since the behavior of some instances has changed." in the release notes
* jgrapht: compile error (KosarajuStrongConnectivityInspector is deprecated already)
* scalaTest: I know there are issues with 3.0.6, although 3.0.7 is published it is not yet mentioned on the website
* selenium: major update, and it looks like a very old version is currently used

I only tested these updates via `sbt test` as don't know much about this project yet!

labra commented:
Thanks...I think the travis build failed temporarily and now it worked. I updated the scala version to 2.12.7...it is possible that other dependencies must be updated also...

Locke commented:
The Travis CI build failed here of course as I added OpenJDK 11 to the build matrix but the Java 11 fix wasn't present in this branch. I just rebased this branch here to resolve the merge conflict and re-trigger the build.

This PR is actually not required but I thought it should help if I update some dependencies.

labra commented:
Indeed it really helps to keep the dependencies updated. Thanks a lot.
